src/terrain/terrain_base.cpp

```diff
@@ -181,8 +113,8 @@ TerrainBase::get_first_not(
                     return z;
                 }
             }
-            return Z_MAX; // -1? should not be minus one, but one should consider that
-                          // this is a possible return value
+            return Z_MAX; // -1? should not be minus one, but one should
```

Comment:
https://en.cppreference.com/w/cpp/utility/expected

Reply:
Its not an error to return Z_MAX, and using it in any other function without testing should not lead to errors.